engine/lib/metastrings.php

```diff
@@ -330,7 +330,7 @@ function _elgg_get_metastring_based_objects($options) {
 		return $dt;
 	} else {
 		$result = get_data_row($query);
-		return $result->calculation;
+		return (int)$result->calculation;
```

Comment:
don't think this is good as calculation can also be float or string as discussed in #1559